File: Backend/src/main/java/com/app/service/InvestorService.java
package com.app.service;

import java.util.List;

import com.app.dto.AuthRequest;
import com.app.dto.BusinessDto;
import com.app.dto.InvestorDto;
import com.app.entities.Investor;

public interface InvestorService {

	String updateInvestor(Long investorId, InvestorDto investor);
	List<InvestorDto> getAllInvestors();
	List<BusinessDto> getInvestorBusinesses(Long investorId);
	String deleteInvestor(Long investorId);
	String createInvestor(InvestorDto investor);
	InvestorDto authenticate(AuthRequest request);
	void updateInvestor(Investor investor);

}

File: FrontEnd/src/service/InvestorService.js
import axios from "axios";
const baseUrl="http://localhost:8080/investor"
class InvestorService{
    //GETTING INVESTOR RESOURCES
    getAllInvestors(){
        return  axios.get(baseUrl+"/investors")
    }
    getAllTrasanctions(){
        return   axios.get(baseUrl+"/transactions")
    }
    getTransactionsForInvestor(investorId){
        return axios.get(baseUrl+"/"+investorId+"/transaction")
    }
    getTransactionsForCampaign(campaignId){
        return    axios.get(baseUrl+"/campaign"+campaignId+"/transaction")
    }
    getInvestorBusinesses(investorId){
        return axios.get(baseUrl+"/"+investorId+"/businesses")
    }
    //UPDATING INVESTOR RESOURCES
    updateInvestor(investor){
        return   axios.put(baseUrl+"/"+investor.id,investor)
    }

    //DELETING INVESTOR RESOURCES
    deleteInvestor(investorId){
        return    axios.delete(baseUrl+"/"+investorId);
    }

    //CREATING INVESTOR RESOURCES
    createInvestor(investor){
        return     axios.post(baseUrl+ "/signup", investor )
    }
    createTransaction(investorId,transaction){
        return axios.post(baseUrl+"/"+investorId+"/transaction"+transaction)
    }
    //LOGIN
    investorLogin(signin){
        return axios.post(baseUrl+"/signin",signin)
    }
}
export default new InvestorService();